### ISSUE TITLE
Null safety

### ISSUE THREAD

**Issue by cloudwebrtc** [flutter-webrtc/flutter-webrtc#458]

**Comment by pan1nx** [comment]
I guess this bug is about null-safety version of the package. Currently, if you add it to a project that has been migrated to null-safety you will get this error:
```bash
Error: Cannot run with sound null safety, because the following dependencies
don't support null safety:

 - package:flutter_webrtc

For solutions, see https://dart.dev/go/unsound-null-safety
```

It would be nice and straight forward to migrate this package also to null-safety. See this: https://dart.dev/null-safety/migration-guide

**Comment by tmthecoder** [comment]
Is there any progress on this? Null-safety is mainstream in flutter now

**Comment by ghenry** [comment]
I've downgraded back to 1.22.6 as sooo many things need to be pinned in overrides. Reported them all though.

**Comment by wer-mathurin** [comment]
I just commit a first attempt in branch (https://github.com/flutter-webrtc/flutter-webrtc/tree/null-safety).
Migrate both the library and the example.  Since the exemple was relying on en external library (Foreground service), I just removed the dependency for now.  You can try it out to find bugs.

**Comment by aytunch** [comment]
@wer-mathurin thanks for the null safety branch. 57 files changed is pretty crazy :)
Will you be merging this to the main repo? We are depending on this package to migrate to Flutter 2.0

**Comment by wer-mathurin** [comment]
I let @cloudwebrtc  decide when he plan to merge it or not.
I would like for people to review it.  

@cloudwebrtc 👍 Maybe we can create a x.x.x-nullsafety package to let the users install and  test it

**Comment by cloudwebrtc** [comment]
@wer-mathurin  It looks like you have done the migration, thank you so much 👍,  I can upload a 0.6.0-null-safety right away.

**Comment by cloudwebrtc** [comment]
@aytunch @wer-mathurin https://pub.dev/packages/flutter_webrtc/versions/0.6.0-nullsafety.0

**Comment by cloudwebrtc** [comment]
fixed v0.6.0
